core/scheduler/resolver/external_resolver.go

```diff
@@ -0,0 +1,69 @@
+package resolver
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/goto/optimus/config"
+	"github.com/goto/optimus/core/scheduler"
+	"github.com/goto/optimus/core/tenant"
+	"github.com/goto/optimus/ext/resourcemanager"
+	"github.com/goto/optimus/internal/errors"
+)
+
+type ExternalOptimusConnectors struct {
+	optimusResourceManagers []OptimusResourceManager
+}
+
+type OptimusResourceManager interface {
+	GetHostURL() string
+	GetJobScheduleInterval(ctx context.Context, tnnt tenant.Tenant, jobName scheduler.JobName) (string, error)
+	GetJobRuns(ctx context.Context, sensorParameters scheduler.JobSensorParameters, criteria *scheduler.JobRunsCriteria) ([]*scheduler.JobRunStatus, error)
+}
+
+const ExternalUpstreamEntity = "ExternalResourceInterface"
+
+// NewExternalOptimusManager creates a new instance of externalResourceResolver
+func NewExternalOptimusManager(resourceManagerConfigs []config.ResourceManager) (*ExternalOptimusConnectors, error) {
```

Comment:
can we reuse `NewExternalUpstreamResolver` on job BC?
we can provide interface specific to this use cases (`GetJobScheduleInterval` and `GetJobRuns`)

Reply:
Then we will be sending request to Job BC from Scheduler BC, even though when the Interface is diffrent. 

what is your reason ?

Reply:
nvm, just want to minimize reimplementation of optimus manager constructor, since it's same with what we have in `core/job/resolver/external_upstream_resolver.go`, turns out it's used for different purpose, we're using same external manager that comes from package `ext`